aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -945,6 +947,26 @@ Tensor _NestedTensor_GeneralizedBMM(const Tensor& self, const Tensor& mat2) {
   return output;
 }
 
+Tensor& matmul_out_nested(const Tensor& tensor1, const Tensor& tensor2, Tensor& result) {
+  // TODO: this is a very quick and dirty implementation
+  //       should improve it to avoid the intermediate memory usage
+  Tensor function_result = at::matmul(tensor1, tensor2);
+  auto function_result_ptr = get_nested_tensor_impl(function_result);
+  std::vector<int64_t> sizes;
```

Comment:
How is the resulting `sizes` different from `function_result_ptr->opt_sizes_` ?

Reply:
Yes they are the same. I did this because `function_result_ptr->opt_sizes_` is private.

Could update if an accessor comes out